Source/WebCore/page/LocalDOMWindow.cpp

```diff
@@ -2807,7 +2807,7 @@ WebCoreOpaqueRoot root(LocalDOMWindow* window)
 CookieStore& LocalDOMWindow::cookieStore()
 {
     if (!m_cookieStore)
-        m_cookieStore = CookieStore::create();
+        m_cookieStore = CookieStore::create(*(this->document()));
```

Comment:
`m_cookieStore = CookieStore::create(document());`

Nothing guarantees that `document()` returns non-null so the CookieStore Factory should take in a `Document*` not a `Document&`